src/geode/model/helpers/convert_to_mesh.cpp

```diff
@@ -539,6 +553,319 @@ namespace
+                    geode::edge_unique_vertices( model, line, line_edge );
+                for( const auto edge_vertex_id :
+                    geode::LRange{ unique_vertices.size() } )
+                {
+                    unique_vertices[edge_vertex_id] =
+                        model2mesh.unique_vertices_mapping.in2out(
+                            unique_vertices[edge_vertex_id] );
+                }
+                const auto solid_edge =
+                    mesh.edges().edge_from_vertices( unique_vertices ).value();
+                model2mesh.line_edges_mapping.map(
+                    { line.id(), line_edge }, solid_edge );
+            }
+        }
+    }
+
+    template < typename Model >
+    std::tuple< std::unique_ptr< geode::SurfaceMesh< Model::dim > >,
+        geode::ModelToMeshMappings >
+        new_convert_model_into_surface( const Model& model )
+    {
+        std::vector<
+            std::reference_wrapper< const geode::SurfaceMesh< Model::dim > > >
+            meshes;
+        meshes.reserve( model.nb_surfaces() );
+        for( const auto& surface : model.surfaces() )
+        {
+            meshes.emplace_back( surface.mesh() );
+        }
+        auto mesh =
+            geode::detail::create_mesh< geode::SurfaceMesh< Model::dim > >(
+                meshes );
+        auto mesh_builder =
+            geode::SurfaceMeshBuilder< Model::dim >::create( *mesh );
+        geode::ModelToMeshMappings model2mesh;
+        build_polygons_from_model( model, *mesh_builder, model2mesh );
+        map_line_edges( model, model2mesh, *mesh );
+        map_corner_vertices( model, model2mesh );
+        mesh_builder->compute_polygon_adjacencies();
+        return std::make_pair( std::move( mesh ), std::move( model2mesh ) );
+    }
+
+    void compute_block_polyhedra_adjacencies(
+        const absl::FixedArray< geode::index_t >& polyhedra,
+        const geode::SolidMesh3D& block_mesh,
+        geode::SolidMeshBuilder3D& mesh_builder )
+    {
+        for( const auto p : geode::Range{ block_mesh.nb_polyhedra() } )
+        {
+            for( const auto f :
+                geode::LRange{ block_mesh.nb_polyhedron_facets( p ) } )
+            {
+                if( const auto adj =
+                        block_mesh.polyhedron_adjacent( { p, f } ) )
+                {
+                    mesh_builder.set_polyhedron_adjacent(
+                        { polyhedra[p], f }, polyhedra[adj.value()] );
+                }
+            }
+        }
+    }
+
+    void build_polyhedra_from_model( const geode::BRep& brep,
+        geode::SolidMeshBuilder3D& mesh_builder,
+        geode::ModelToMeshMappings& brep2mesh )
+    {
+        for( const auto& block : brep.blocks() )
+        {
+            const auto& block_mesh = block.mesh();
+            absl::FixedArray< geode::index_t > polyhedra(
+                block_mesh.nb_polyhedra() );
+            for( const auto p : geode::Range{ block_mesh.nb_polyhedra() } )
```

Comment:
Here, p, v and f are used all through the function -> use variables with at least 3 letters, it makes it safer and easier to read the code later by being able to search the code for occurences of the variables